examples/tutorials/subplots.py

```diff
@@ -0,0 +1,244 @@
+# also possible to use a question mark ``"?"`` to let GMT decide automatically
+# on what is the most appropriate width/height for the each subplot's map
+# frame.
+
+###############################################################################
+# .. tip::
+#
+#     In the above example, we used the following commands to activate the
+#     four subplots explicitly one after another::
+#
+#         fig.basemap(..., panel=[0, 0])
+#         fig.basemap(..., panel=[0, 1])
+#         fig.basemap(..., panel=[1, 0])
+#         fig.basemap(..., panel=[1, 1])
+#
+#     In fact, we can just use ``fig.basemap(..., panel=True)`` without
+#     specifying any subplot index number, and GMT will automatically activate
+#     the next subplot panel.
+
+###############################################################################
+# .. note::
+#
+#     All plotting functions (e.g. :meth:`pygmt.Figure.coast`,
+#     :meth:`pygmt.Figure.text`, etc) are able to use ``panel`` argument when
+#     in subplot mode. Once a panel is activated using ``panel`` or
+#     :meth:`pygmt.Figure.set_panel`, subsequent plotting commands that don't
+#     set a ``panel`` will have their elements added to the same panel as
+#     before.
+
+###############################################################################
+# Shared X and Y axis labels
+# --------------------------
+# In the example above with the four subplots, the two subplots for each row
+# have the same Y-axis range, and the two subplots for each column have the
+# same X-axis range. You can use the ``sharex``/``sharey`` arguments to set a
+# common X and/or Y axis between subplots.
+
+fig = pygmt.Figure()
+with fig.subplot(
+    nrows=2,
+    ncols=2,
+    figsize=("15c", "6c"),  # width of 15 cm, height of 6 cm
+    autolabel=True,
+    margins=["0.3c", "0.2c"],  # horizontal 0.3 cm and vertical 0.2 cm margins
+    title="My Subplot Heading",
+    sharex="b",  # shared x-axis on the bottom side
+    sharey="l",  # shared y-axis on the left side
+    frame="WSrt",
+):
+    fig.basemap(region=[0, 10, 0, 10], projection="X?", panel=True)
+    fig.basemap(region=[0, 20, 0, 10], projection="X?", panel=True)
+    fig.basemap(region=[0, 10, 0, 20], projection="X?", panel=True)
+    fig.basemap(region=[0, 20, 0, 20], projection="X?", panel=True)
+fig.show()
+
+###############################################################################
+# ``sharex="b"`` indicates that subplots in a column will share the x-axis, and
+# only the **b**\ ottom axis is displayed. ``sharey="l"`` indicates that
+# subplots within a **R**\ ow will share the y-axis, and only the **l**\ eft
+# axis is displayed.
+#
+# Of course, instead of using the **sharex**/**sharey** option, you can also
+# set a different **frame** for each subplot to control the axis properties
+# individually for each subplot.
+
+###############################################################################
```

Comment:
Considering the bugs and the tricks, do you think we should call `subplot` twice to get the same advanced layout? 
```python
import pygmt
fig = pygmt.Figure()

with fig.subplot(nrows=1, ncols=2, figsize=("15c", "6c"), autolabel="b)"):
    fig.basemap(
        region=[0, 5, 0, 5], projection="X?", frame=["af", "WSne"], panel=[0, 0]
    )
    fig.basemap(
        region=[0, 5, 0, 5], projection="X?", frame=["af", "WSne"], panel=[0, 1]
    )
    
with fig.subplot(nrows=1, ncols=1, figsize=("15c", "6c"), autolabel="a)", yshift="h+2c"):
    fig.basemap(
        region=[0, 10, 0, 10], projection="X?", frame=["af", "WSne"], panel=[0, 0]
    )
    fig.text(text="TEXT", x=5, y=5)

fig.show()
```

Reply:
I was actually thinking whether to take this section out. But your code example looks good so I think I'll switch to yours. Will edit the documentation notes too.